test/unit/ObjectMakrTests/processFunctions.js

```diff
@@ -0,0 +1,85 @@
+define(["mocks"], function (mocks) {
+    return function () {
+        var expect = require("chai").expect;
+
+        it("has three functions", function () {
+            // Arrange
+            var objectMaker = mocks.mockObjectMakr();
+
+            var inheritance = {
+                Organism: {
+                    Animal: {},
+                    Plant: {}
+                }
+            };
+
+            // Act
+            objectMaker.functions = {};
+            objectMaker.processFunctions(inheritance, Object, "Object");
+            var actualMapping = objectMaker.functions;
+
+            // Assert
+            expect(Object.keys(actualMapping).length).to.equal(3);
```

Comment:
What if I screwed up writing ObjectMakr and the names ended up being `["Organismnope", "Animalnope", "Plantnope"]`?